src/Raven.Server/Smuggler/Documents/StreamDestination.cs

```diff
@@ -54,6 +55,23 @@ public IDisposable Initialize(DatabaseSmugglerOptionsServerSide options, Smuggle
             _writer = new BlittableJsonTextWriter(_context, _gzipStream);
             _options = options;
 
+            var skipCountersMetadata = _options.OperateOnTypes.HasFlag(DatabaseItemType.CounterGroups) == false;
+            var skipAttachmentsMetadata = _options.OperateOnTypes.HasFlag(DatabaseItemType.Attachments) == false;
+
+            _filterMetadataProperty = metadataProperty =>
+            {
+                switch (metadataProperty)
```

Comment:
add 

```
if (skipCountersMetadata == false && skipAttachmentsMetadata  == false)
    return false;
```

This way you will avoid doing string comparisons for each metadata property if both of them are `false`

Reply:
done